### ISSUE TITLE
Authentication changes in Resources?

### ISSUE THREAD

**Issue by RVN-BR** [Luracast/Restler#443]
Hi All,

I wonder if there has been further changes in the authentication of the resrouces/apidoc part?  I havent migrated to the new swagger version which is served up automatically (I tried actually, but then rverted back due to some reasons, including we dont serve static content together with dynamic, etc, etc)...

Our authentication is quite complex, and it involves a few levels, such as: 
Users (with different permissions)
Apps (with different permissions)
Internal methods (not called via web)
Public methods (access without authentication)

The authentication seems to be stable, and is taken care of by the isAllowed method of our auth class.  The way we define viewing levels and required permissions is with a docblock on the function level. such as:

```php
    /* 
    * @class MyNamespace\Core\Authentication {@apiType public} 
    */
	function getPublic() {}
```
or
```php
    /* 
    * @class MyNamespace\Core\Authentication {@apiType app} {@appPermission read,write} 
    */
	function getAppRW() {}
``` 
(above for illustration)...

Usually I would add a @protected notation to the top of every class, and by returning a true on the mthods that had apiType public when they were passed to isAllowed and to the verifyAccess (resource authentication class) I was able to get it working as intended...

Basically @protected would make everything run through the authentication code, and my custom variables would be fed to the auth methods and that would show them or not.  This simply stopped working...

I also noticed a lot of strict errors/warnings which I had to suppress in Resources.php line 1000 in reference to calling my authentication class for resources, so maybe that has something to do with it?

The API authentication seems to be solid, still allowing public methods to be called even with the @protected notation, but they are hidden from the API Docs...  If I try "show protected methods" I get the inverse problem, I see the public methods, but I also see the methods that I dont have access to...

My idea is to show the developer only what he/she has access to based on the credentials they already have loaded (or public methods only if they dont have any credentials)... this is on php 5.6.7

**Comment by Arul-** [comment]
you are using RC5 or RC6?

**Comment by RVN-BR** [comment]
RC5.... I tried RC6 but had problems and came back to RC5 after we had some discussions a while back that RC6 breaking changes still werent documented/ready, I dont know if you will recall... this is why its strange... I'm fixed on 3.0.0rc5 on composer... it shouldnt have upgraded anything...

**Comment by Arul-** [comment]
Explorer in RC6 is still Work in Progress. We will be migrating to Swagger 2 spec soon!
